iceoryx_binding_c/test/test.hpp

```diff
@@ -17,6 +17,8 @@
 #ifndef IOX_DDS_GATEWAY_TEST_TEST_HPP
 #define IOX_DDS_GATEWAY_TEST_TEST_HPP
 
+#include "iceoryx_hoofs/testing/mocks/error_handler_mock.hpp"
```

Comment:
Why do we actually require an error handler mock? We can always set a temporary error handler and check if it was called with the correct arguments?

Reply:
The `ErrorHandlerMock` contains call code which is only needed in test code. Now, `ErrorHandler` on the other hand, contains the code needed during runtime of an iceoryx system and is not cluttered with test-only code anymore. The templatization of `setTemporaryErrorHandler` is necessary to provide a user friendly way to use a different error enums for each module (posh, C binding).